interfaces/apparmor/spec.go

```diff
@@ -181,7 +186,9 @@ func (spec *Specification) AddLayout(si *snap.Info) {
 			fmt.Fprintf(&buf, "  %s rw,\n", path)
 			WritableProfile(&buf, path, 2) // At least / and /some-top-level-directory
 		}
-		spec.AddUpdateNS(buf.String())
+		for _, line := range strings.SplitAfter(buf.String(), "\n") {
```

Comment:
It seems a bit wasteful to construct bytes.Buffer and append lines and then split it back; can we simply collect them in an array and avoid temporary bytes.Buffer?

Reply:
That's true but I wanted to get feedback on the proposal first. The whole buffer can go away and we can simply add each snippet as they are created.

Reply:
This is now addressed.